SAMLPortal/Controllers/API/SetupController.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Novell.Directory.Ldap;
+using SAMLPortal.Models;
+
+namespace SAMLPortal.Controllers.API
+{
+	[Route("api/setup")]
+	[ApiController]
+	//[ServiceFilter(typeof(SetupAsyncActionFilter))]
+	//[Produces("application/json")]
+	public class SetupController : Controller
+	{
+		public SetupController()
+		{
+		}
+
+		// GET api/setup
+		[HttpPost]
+		[Route("getResultsFromFilters")]
+		public ActionResult<IEnumerable<string>> GetResultsFromFilters([FromBody]string testUsername)
+		{
+			if (GlobalSettings.GetInt("CONFIG_SETUPASSISTANT_STEP") >= 5)
+			{
+				try
+				{
+					LdapConnection connection = new LdapConnection();
+					connection.SecureSocketLayer = bool.Parse(GlobalSettings.Get("LDAP_SSL"));
+
+					connection.Connect(GlobalSettings.Get("LDAP_Host"), (int)GlobalSettings.GetInt("LDAP_Port"));
+					connection.Bind(GlobalSettings.Get("LDAP_BindDN"), GlobalSettings.Get("LDAP_BindPass"));
+
+					var userSearchFilter = string.Format(GlobalSettings.Get("LDAP_UsersFilter"), testUsername);
+					var adminSearchFilter = string.Format(GlobalSettings.Get("LDAP_AdminFilter"), testUsername);
+
+					List<string> filters = new List<string> { userSearchFilter, adminSearchFilter };
+					Dictionary<string, List<string>> allResults = new Dictionary<string, List<string>>();
+
+					var filtersEnumerator = filters.GetEnumerator();
+					for (int i = 0; filtersEnumerator.MoveNext(); i++)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This loop's stop condition tests 'filtersEnumerator' but the incrementer updates 'i'.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)